tensorflow/core/ops/nn_ops.cc

```diff
@@ -301,6 +301,36 @@ REGISTER_OP("FusedBatchNormGradV3")
     .Attr(GetConvnetDataFormat2D3DAttrString())
     .Attr("is_training: bool = true")
     .SetShapeFn(shape_inference::FusedBatchNormGradShape);
+
+REGISTER_OP("_FusedBatchNormGradEx")
+    .Input("y_backprop: T")
+    .Input("x: T")
+    .Input("scale: float")
+    .Input("reserve_space_1: U")
+    .Input("reserve_space_2: U")
+    .Input("reserve_space_3: U")
+    .Input("offset: float")
+    .Input("y: T")
+    .Output("x_backprop: T")
+    .Output("scale_backprop: U")
+    .Output("offset_backprop: U")
+    .Output("reserve_space_4: U")
+    .Output("reserve_space_5: U")
+    .Output("side_input_backprop: num_side_inputs * T")
+    .Attr("T: {half, float}")
+    .Attr("U: {float}")
+    .Attr("epsilon: float = 0.0001")
+    .Attr("num_side_inputs: int >= 0 = 0")
+    .Attr("activation_mode: string = \"Identity\"")
+    .Attr(GetConvnetDataFormat2D3DAttrString())
+    .Attr("is_training: bool = true")
```

Comment:
I understand is_training is present in the ops above already, but I'm still a bit confused what does this attribute mean for the op definition?

Reply:
Yes, the attribute is from the above op `FusedBatchNormGradV3`. Both the `FusedBatchNormGradV3` and this new `_FusedBatchNormGradEx` shared the same base class `FusedBatchNormGradOpBase` which needs this attribute to determine to call normal `functor::FusedBatchNormGrad` or `functor::FusedBatchNormFreezeGrad`. And we don't expect `is_training=False` for `_FusedBatchNormGradEx` and only check it.